File: domain/src/main/java/com/balv/imdb/domain/usecases/GetNextMoviePageUseCase.java
package com.balv.imdb.domain.usecases;


import com.balv.imdb.domain.models.ApiResult;

import javax.inject.Inject;

import io.reactivex.Observable;

public class GetNextMoviePageUseCase extends BaseUseCase<Integer, ApiResult> {

    @Inject
    public GetNextMoviePageUseCase(){};

    @Override
    public Observable<ApiResult> execute(Integer page){
        return mMovieRepo.getNextRemoteDataPage(page);
    }
}

File: domain/src/main/java/com/balv/imdb/domain/usecases/GetNextMoviePageUseCase.kt
package com.balv.imdb.domain.usecases

import com.balv.imdb.domain.models.ApiResult
import com.balv.imdb.domain.repositories.IMovieRepository
import io.reactivex.Observable
import javax.inject.Inject


class GetNextMoviePageUseCase @Inject constructor(
    mMovieRepo: IMovieRepository
) : BaseUseCase<Int, ApiResult<*>?>(mMovieRepo) {
    override suspend fun execute(input: Int): ApiResult<*> {
        return mMovieRepo.getNextRemoteDataPage(input)
    }
}
